File: 979-di-string-match/di-string-match.cpp
class Solution {
public:
    vector<int> diStringMatch(string s) {
        int n = s.length();
        int i=0 , j=n;
        vector<int> arr(n+1);
        for(int x=0 ; x<n ; x++){
            if(s[x] == 'I') arr[x] = i++;
            else arr[x] = j--;
        }
        arr[n] = j;
        return arr;
    }
};

File: 979-di-string-match/di-string-match.py
class Solution(object):
    def diStringMatch(self, s):
        n = len(s)
        i , j = 0 , n
        arr = [0] * (n+1)
        for x in range(n):
            if(s[x] == 'I'):
                arr[x] = i
                i += 1
            else :
                arr[x] = j
                j -= 1
        arr[n] = j
        return arr